packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,142 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+
+type AsOrChildren =
+  | {
+      as: ExoticComponent<any> | ComponentType<any>;
+      children?: never;
+    }
+  | {
+      as?: never;
+      children: ReactElement | ReactElement[];
+    };
+
+type IconInternalProps = {
+  /**
+   * The name prop is used to check if "trimmed" styling should be applied,
+   * which sets the `width` to `auto`.
+   */
+  name?: string;
+  /**
+   * Determines the size of the icon
+   */
+  size?: 'large' | 'medium' | 'small' | 'tiny';
+  /**
+   * Determines the fill color used
+   */
+  variant?: IconVariant;
+  /**
+   * SVG viewBox attribute
+   */
+  viewBox?: `${number} ${number} ${number} ${number}`;
+} & AsOrChildren;
+
+export type IconProps = IconInternalProps &
+  CommonProps &
+  SVGAttributes<SVGSVGElement>;
+
+const styles = {
+  large: css({
+    height: '32px',
+    width: '32px',
+  }),
+  medium: css({
+    height: '24px',
+    width: '24px',
+  }),
+  muted: css({
+    fill: tokens.colorTextLight,
+  }),
+  negative: css({
+    fill: tokens.colorNegative,
+  }),
+  positive: css({
+    fill: tokens.colorPositive,
+  }),
+  root: css({
+    display: 'inline-block',
+    fill: tokens.colorBlueBase,
+  }),
+  secondary: css({
+    fill: tokens.colorContrastMid,
+  }),
+  small: css({
+    height: '18px',
+    width: '18px',
+  }),
+  tiny: css({
+    height: '16px',
+    width: '16px',
+  }),
+  trimmed: css({
+    width: 'auto',
+  }),
+  warning: css({
+    fill: tokens.colorWarning,
+  }),
+  white: css({
+    fill: tokens.colorWhite,
+  }),
+};
+
+export const Icon = forwardRef<SVGSVGElement, IconProps>(function Icon(
+  {
+    as,
+    children,
+    className,
+    name,
+    variant = 'primary',
+    size = 'small',
+    testId = 'cf-ui-icon',
+    viewBox = '0 0 24 24',
+    ...otherProps
+  },
+  forwardedRef,
+) {
+  const style = css`
```

Comment:
I think we should have a function that accepts all icon props and that returns a css class that needs to be applied.

`css({})` has to used only once so we don't create many rules beforehand and styles is as performant as it can be